presto-main/src/main/java/com/facebook/presto/cost/HistoryBasedPlanStatisticsTracker.java

```diff
@@ -144,6 +149,16 @@ public Map<PlanNodeWithHash, PlanStatisticsWithSourceInfo> getQueryStats(QueryIn
                 double nullJoinBuildKeyCount = planNodeStats.getPlanNodeNullJoinBuildKeyCount();
                 double joinBuildKeyCount = planNodeStats.getPlanNodeJoinBuildKeyCount();
 
+                JoinNodeStatistics joinNodeStatistics = JoinNodeStatistics.empty();
+                if (planNode instanceof JoinNode) {
+                    joinNodeStatistics = new JoinNodeStatistics(Estimate.of(nullJoinBuildKeyCount), Estimate.of(joinBuildKeyCount));
+                }
+
+                TableWriterNodeStatistics tableWriterNodeStatistics = TableWriterNodeStatistics.empty();
+                if (isScaledWriterStage && planNode instanceof TableWriterNode) {
```

Comment:
is TableWriterMergeNode relevant here?

Reply:
No, the scaled writer optimization is only related to table writer node, not related to table writer merger node.